idm/idm.go

```diff
@@ -0,0 +1,73 @@
+// Package idm manages resevation/release of numerical ids from a configured set of contiguos ids
+package idm
+
+import (
+	"fmt"
+
+	"github.com/docker/libnetwork/bitseq"
+	"github.com/docker/libnetwork/datastore"
+)
+
+// Idm manages the reservation/release of numerical ids from a contiguos set
+type Idm struct {
+	start  int
```

Comment:
Can you please change the idm start, end and the return values of `GetID` and other apis to uint32?